### PR TITLE
Use PackageName type

### PR DESCRIPTION
**Pull request by shayne-fletcher**
As suggested here https://github.com/digital-asset/daml/pull/4641#pullrequestreview-362615364, change `Maybe String` to `Maybe PackageName`.

### CODE REVIEW

File: language-support/ts/codegen/src/TsCodeGenMain.hs [hunk 1]
```diff
@@ -47,36 +47,38 @@ optionsParserInfo = info (optionsParser <**> helper)
...
       dalfs <- either fail pure $ DAR.readDalfs archive
       DAR.DalfManifest{packageName} <- either fail pure $ DAR.readDalfManifest archive
+      packageName <- pure (PackageName . T.pack <$> packageName)
       forM ((DAR.mainDalf dalfs, packageName) : map (, Nothing) (DAR.dalfs dalfs)) $
         \(dalf, mbPkgName) -> do
           (pkgId, pkg) <- either (fail . show)  pure $ Archive.decodeArchive Archive.DecodeAsMain (BSL.toStrict dalf)
           pure (pkgId, (pkg, mbPkgName))
 
-mergePackageMap :: [(PackageId, (Package, Maybe String))] -> Either String (Map.Map PackageId (Maybe String, Package))
+mergePackageMap :: [(PackageId, (Package, Maybe PackageName))] -> Either String (Map.Map PackageId (Maybe PackageName, Package))
```
Comment (cocreature) [inline language-support/ts/codegen/src/TsCodeGenMain.hs]: Might be slightly simpler if you go for `Either Text` and then do the conversion at the call site (or just use `Data.Text.IO.putStrLn`).

Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: Ever so slightly. Did that.

File: language-support/ts/codegen/src/TsCodeGenMain.hs [hunk 2]
```diff
@@ -85,19 +87,19 @@ main = do
...
     case mergePackageMap ps of
-      Left err -> fail err
+      Left err -> fail . T.unpack $ err
       Right pm ->
         forM_ (Map.toList pm) $
         \(pkgId, (mbPkgName, pkg)) -> do
           let id = show $ unPackageId pkgId
-          let name = fromMaybe id mbPkgName
-          let asName = if name == id then "itself" else name
+              name = maybe id (T.unpack . unPackageName) mbPkgName
```
Comment (hurryabit) [inline language-support/ts/codegen/src/TsCodeGenMain.hs]: Why did you remove the `let`s? OCaml's `let x = ... in let y = ... in ...` is superior to Haskell's `let { x = ...; y = ... } in ...`, which is basically a huge footgun.

Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: We had,
```
  let x = ...
  let y = ...
  ...
```
in a monadic context. I replaced with,
```
let x = ...
     y = ...
    ....
```
as I believed it to be more idiomatic? You want I restore it?

Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: Let me know if so and I'll do so in a follow up PR!

Comment (hurryabit) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: I don't know which one is more idiomatic. Let's go with the safer version then, i.e., the old version.
